android/src/main/java/io/ably/flutter/plugin/dto/EnrichedConnectionStateChange.java

```diff
@@ -0,0 +1,15 @@
+package io.ably.flutter.plugin.dto;
+
+import io.ably.lib.realtime.ConnectionStateListener;
+
+public class EnrichedConnectionStateChange {
```

Comment:
`EnrichedConnectionStateChange ` can be redefined to extend `ConnectionStateListener.ConnectionStateChange`

Reply:
That's not a good idea and the problem of ably-java implementation. This class `ConnectionStateListener.ConnectionStateChange` definitely should be final, it's internal data class, there is no reason somebody should extend it.

Reply:
Why not to add those properties in `ConnectionStateListener.ConnectionStateChange` directly?